layout-invalidate 동작 구조 개선

YSYoon opened the pull request:
기존 동작구조상 필요 이상으로 `View.requestLayout()`이 호출되고 있으며
그로 인해 필요 이상으로 빈번하게 `View.invalidate()`가 발생하고 있습니다.

일차적으로 `5e7f833` 커밋을 반영한 PP1 펌웨어로 테스트해본 결과
* 페이지 넘김 및 확대/축소 시 반응성개선,
* 페이지 넘길 때의 전류 소모량이 감소되는 결과를 확인할 수 있었습니다

YSYoon committed:
ImageView를 custom View로 대체

ImageView.setImageBitmap(Bitmap) 호출 시 이미지 크기가 바뀌면 내부적으로
requestLayout -> invalidate 싸이클이 돌게 되는데, 페이지 이동/확대축소 시에
반복적으로 null로 세팅하고 새 Bitmap을 넣어주게 되면서 부하가 커져 화면 갱신이 지연되고
전류 소모량이 증가하는 문제가 있어 layout갱신을 제어할 수 있는 별도 custom View를 적용함

YSYoon committed:
확대/축소 시 동작 추가 개선

확대/축소 진행 중에 실제 Bitmap 설정이 완료되지 않은 경우 굳이 invalidate() 를 진행할 필요 없을 것으로 판단되어 스킵
꼭 필요하지 않은 requestLayout() 제거

YSYoon committed:
PageContentView 의 requestLayout 정리

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+    @ColorInt private int paperColor;
+    private ImageView.ScaleType scaleType;
+    private Rect rect = new Rect();
+    private boolean isHighQualityView;
```
`is`는 (지금은 없지만) getter method에 붙이고 boolean field 이름앞에는 붙이지 않습니다.

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+    private boolean didDrawOnce;
+
+    PageContentImageView(Context context) {
+        this(context, null);
```
`this(context, false)` 를 호출해야 좀 더 자연스러운 overloading일 것 같습니다.

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+    private boolean bitmapSet;
+    private Bitmap bitmap;
+    @ColorInt private int paperColor;
+    private ImageView.ScaleType scaleType;
```
이거는 의미가 없는 것 같네요.

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+
+    public void setImageBitmap(Bitmap bitmap) {
+        if (this.bitmap != null) {
+            this.bitmap.recycle();
```
여기서 bitmap을 임의로 recycle하는건 외부에서 의도하지 않은 동작이 될 것 같습니다.

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+            this.bitmap.recycle();
+        }
+        this.bitmap = bitmap;
+        this.bitmapSet = (bitmap != null);
```
`bitmapSet`은 그냥 `bitmap != null` 로 대체하면 될 것 같네요.

freekering commented on src/main/java/com/ridi/books/viewer/reader/pagecontent/PageContentImageView.java:
```diff
@@ -0,0 +1,80 @@
+        }
+        this.bitmap = bitmap;
+        this.bitmapSet = (bitmap != null);
+        this.didDrawOnce = false;
```
`didDrawOnce` 보다는 반대 의미의 `dirty`가 좀 더 와닿는 이름일 것 같아 제안드립니다.

freekering commented:
@YSYoon 대체적으로 불필요한 `this.` 는 생략하고 있습니다.

YSYoon committed:
코드리뷰 반영

불필요한 this 제거, 불필요한 변수 제거, import 정리
PageContentImageView 에서 임의로 Bitmap.recycle() 하던 코드 제거
